sbi/src/attestation.rs

```diff
@@ -5,29 +5,149 @@
-    OpenDice = 2,
+/// The maximum number of measurement registers in the attestation capabilities.
+pub const MAX_MEASUREMENT_REGISTERS: usize = 32;
+
+flags! {
+    /// Attestation evidence formats.
+    #[derive(Default)]
+    #[repr(u8)]
+    pub enum EvidenceFormat: u8 {
+        /// Single layer DICE TCB
+        /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
+        #[default]
+        DiceTcbInfo = 1,
+        /// X.509 extension for multiple layers DICE TCB.
+        /// https://trustedcomputinggroup.org/resource/dice-attestation-architecture/
+        DiceMultiTcbInfo = 2,
+        /// Open DICE profile
+        /// https://pigweed.googlesource.com/open-dice/+/HEAD/docs/specification.md
+        OpenDice = 4,
+    }
+}
+
+/// A list of supported hash algorithms.
+#[derive(Copy, Clone, Default, Debug)]
+#[repr(u8)]
+pub enum HashAlgorithm {
+    /// SHA-384
+    #[default]
+    Sha384 = 1,
+    /// SHA-512
+    Sha512 = 2,
+}
+
+/// Attestation Capabilities
+///
+/// This structure exposes the supported attestation capabilities to the SBI
+/// GetCapabilities caller. It lets the caller know which hash algorithms,
+/// evidence formats, and measurements mappings the SBI implementation supports.
+#[repr(C)]
+#[derive(Default)]
+pub struct AttestationCapabilities {
+    /// The TCB Secure Version Number.
+    pub tcb_svn: u64,
+    /// The supported hash algorithm.
+    pub hash_algorithm: HashAlgorithm,
+    /// The supported evidence formats. This is a bitmap.
+    pub evidence_formats: FlagSet<EvidenceFormat>,
+    /// Number of static measurement registers.
+    pub static_measurements: u8,
+    /// Number of runtime measurement registers.
+    pub runtime_measurements: u8,
+    /// Array of all measurement register descriptors.
+    pub measurement_registers: [MeasurementRegisterDescriptor; MAX_MEASUREMENT_REGISTERS],
+}
+
+impl AttestationCapabilities {
+    /// Create new attestation capabilities structure.
+    pub fn new(
+        tcb_svn: u64,
+        hash_algorithm: HashAlgorithm,
+        evidence_formats: impl Into<FlagSet<EvidenceFormat>>,
+        static_measurements: u8,
+        runtime_measurements: u8,
+    ) -> Self {
+        AttestationCapabilities {
+            tcb_svn,
+            hash_algorithm,
+            evidence_formats: evidence_formats.into(),
+            static_measurements,
+            runtime_measurements,
+            ..Default::default()
+        }
+    }
+
+    /// Add a measurement register to the attestation capabilities.
+    pub fn add_measurement_register(
+        &mut self,
+        register: MeasurementRegisterDescriptor,
+        index: usize,
+    ) -> Result<&mut AttestationCapabilities> {
+        // We do not allow for a sparse measurement register array.
+        if index + 1 > (self.static_measurements + self.runtime_measurements) as usize {
+            return Err(Error::Overflow);
+        }
```

Comment:
@abrestic-rivos FYI I added this check after you approved the PR.
We want to make sure the SBI implementation does not create a sparse array of measurement registers.